### PR TITLE
Use DB-specific nomenclature

### PR DESCRIPTION
**Pull request by dogversioning**
This PR makes the following changes:
- use a given database's nomenclature for things like schema/database #258
- removes nested dirs in module_allowlist #239
- Moves dangling bit of db config from main into databases proper

### Checklist
- [X] Consider if documentation (like in `docs/`) needs to be updated
- [X] Consider if tests should be added
- [X] Update template repo if there are changes to study configuration

### CODE REVIEW

File: cumulus_library/databases/duckdb.py [hunk 1]
```diff
@@ -22,9 +22,14 @@
...
 
-    def __init__(self, db_file: str):
+    def __init__(self, db_file: str, schema_name: str | None = None):
         super().__init__("main")
         self.db_type = "duckdb"
+        # As of the 1.0 duckdb release, local scopes, where schema names can be provided
+        # as configuration to duckdb.connect, are not supported.
+        # https://duckdb.org/docs/sql/statements/set.html#syntax
+
+        # This is where the connection config would be supplied when it is supported
```
Comment (dogversioning) [inline cumulus_library/databases/duckdb.py:32]: I'm electing to set up the schema stuff here, and in the DB utils call, even though the local module stuff is not configured yet, mostly just as breadcrumbs for the future.

File: cumulus_library/cli_parser.py [hunk 2]
```diff
@@ -69,11 +69,16 @@ def add_db_config(parser: argparse.ArgumentParser, input_mode: bool = False) ->
...
-        dest="schema_name",
-        help="Database name (for Athena) or file (for DuckDB)",
+        help="Database/schema name (for Athena) or file (for DuckDB)",
+    )
+    group.add_argument(
+        "--schema_name",
+        # In Athena, we use this as the schema_name (which is also called a Database
+        # in their UX).
+        #
+        # In DuckDB, we use this as the path to the filename to store tables.
```
Comment (mikix) [inline cumulus_library/cli_parser.py:79]: copy/paste error

File: main
Comment (mikix) [inline main:1]: :scissors:?

File: tests/test_databases.py [hunk 3]
```diff
@@ -237,6 +237,11 @@ def test_pyarrow_types_from_sql(db, data, expected, raises):
             databases.AthenaDatabaseBackend,
             does_not_raise(),
         ),
+        (
+            {**{"db_type": "athena", "database": "testtwp"}, **ATHENA_KWARGS},
```
Comment (mikix) [inline tests/test_databases.py:241]: Maybe also a test where database and schema_name are equal?

**Review by mikix — APPROVED** [review]

File: main
Comment (dogversioning) [reply main:1]: think this came over from the other branch i was debugging.

**Pull request merged**